server/src/main/java/org/opensearch/gateway/TransportNodesListGatewayStartedShardsBatch.java

```diff
@@ -264,13 +269,26 @@ public Map<ShardId, GatewayStartedShard> getNodeGatewayStartedShardsBatch() {
 
         public NodeGatewayStartedShardsBatch(StreamInput in) throws IOException {
             super(in);
-            this.nodeGatewayStartedShardsBatch = in.readMap(ShardId::new, GatewayStartedShard::new);
+            this.nodeGatewayStartedShardsBatch = in.readMap(ShardId::new, i -> {
+                if (i.readBoolean()) {
```

Comment:
what is this boolean check? same for StreamOutput below

Reply:
Map of ShardId : GatewayStartedShard - can also have null entries. So, this check clarifies if that value is null or not. Based on that we do ser-deser.